### PR TITLE
t: Extract helper for client user agent initialization

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Review by kalikiana — APPROVED** [review]
The PR needs a better name. Like `Unify use of OpenQA::Client with Test:Mojo`. Otherwise it sounds like it's about tests for the client.

That said, I like it

**Comment by okurz** [comment]
ok, updated the PR description to use the actual git commit subject message from the last commit because this one describes the most important, relevant change for this PR. The other changes are merely trivial things going along with it.

I have a specific question. Right now I named the new helper "client" but I wonder if the test code reads nicely with this name. Are you ok with the choice or have a better idea?

**Comment by kalikiana** [comment]
> ok, updated the PR description to use the actual git commit subject message from the last commit because this one describes the most important, relevant change for this PR. The other changes are merely trivial things going along with it.
> 
> I have a specific question. Right now I named the new helper "client" but I wonder if the test code reads nicely with this name. Are you ok with the choice or have a better idea?

I was pondering that but couldn't think of a better name. I think if we use it consistently as `client($t)` it's clear enough.

File: t/lib/OpenQA/Test/Client.pm [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+use Test::Mojo;
+use OpenQA::Client;
+
+use Exporter 'import';
+
+our @EXPORT = qw(client);
+
+# setup test application with API access
+# note: Test::Mojo looses its app when setting a new ua (see https://github.com/kraih/mojo/issues/598).
+sub client {
```
Comment (Martchus) [inline t/lib/OpenQA/Test/Client.pm:12]: I would have preferred adding this to the test utilities as it seems unlikely that related functions are added into this new module later. I would have also preferred a name like `create_client` for a function which actually creates a new instance and not just returns an existing one.

Comment (okurz) [reply t/lib/OpenQA/Test/Client.pm:12]: yes, this is why this PR was open for 7 days and I asked for feedback.

**Pull request merged**